background/pool.ts

```diff
@@ -72,71 +73,62 @@ export class RequestStoragePool implements RequestPool {
     throw new Error(`Unknown request type ${request}`)
   }
 
-  private waitForTransaction(request: PendingTransactionRequest) {
+  private waitForTransaction(request: StorageTransactionRequest) {
     return new Promise<HexString>((resolve, reject) => {
-      const subscriber = async ({ account }: State) => {
-        const txLog = account[request.accountId].transactionLog[request.id]
+      const subscriber = async (state: State) => {
+        const log = new StateActor(state).getTransactionLog(request.id)
 
         // Bundler is still processing this user operation
-        if (txLog.status === TransactionStatus.Sent) {
+        if (log.status === TransactionStatus.Sent) {
           return
         }
 
         this.storage.unsubscribe(subscriber)
+        this.storage.set((state) => {
+          delete state.requestLog[request.id]
+        })
```

Comment:
Does this suggest that `rejected` transactions might no longer be kept and therefore won't show on the `Activity` page in the Popup?

Reply:
Yes, according to the design, rejected transactions would not be shown in activity, so we could skip them entirely in storage to also save storage space.

Reply:
Got it, thanks!